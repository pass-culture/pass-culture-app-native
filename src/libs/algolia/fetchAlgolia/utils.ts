export const buildHitsPerPage = (hitsPerPage: number | null) =>
  hitsPerPage ? { hitsPerPage } : null
