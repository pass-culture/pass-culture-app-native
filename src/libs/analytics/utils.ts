import { NativeSyntheticEvent, NativeScrollEvent } from 'react-native'

type Props = NativeSyntheticEvent<NativeScrollEvent>['nativeEvent'] & { padding?: number }

export const isCloseToBottom = ({
  layoutMeasurement,
  contentOffset,
  contentSize,
  padding = 20,
}: Props) => layoutMeasurement.height + contentOffset.y >= contentSize.height - padding

export const isCloseToEndHorizontal = ({
  layoutMeasurement,
  contentOffset,
  contentSize,
  padding = 0,
}: Props) => layoutMeasurement.width + contentOffset.x >= contentSize.width - padding

// We don't send integers to firebase because they will be cast into int_value, float_value,
// or double_value in BigQuery depending on its value. To facilitate the work of the team,
// we just cast it to string.
export const prepareLogEventParams = (params: Record<string, unknown>) =>
  Object.keys(params).reduce((acc: Record<string, unknown>, key) => {
    acc[key] = typeof params[key] === 'number' ? (params[key] as number).toString() : params[key]
    return acc
  }, {})
