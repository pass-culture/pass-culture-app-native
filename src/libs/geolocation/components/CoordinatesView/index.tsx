import React, { FunctionComponent } from 'react'
import { StyleSheet, Text, View, StyleProp, ViewStyle } from 'react-native'
import { GeoCoordinates } from 'react-native-geolocation-service'

// eslint-disable-next-line no-restricted-imports
import { ColorsEnum } from 'ui/theme/colors'

interface Props {
  position: GeoCoordinates | null
  style?: StyleProp<ViewStyle>
}

const styles = StyleSheet.create({
  container: { flexDirection: 'row', flex: 1 },
  geolocationText: { fontSize: 9 },
  coordinateLabel: { color: ColorsEnum.ACCENT },
})

/**
 * Temporary component to demonstrate the geolocation feature
 */
const CoordinatesView: FunctionComponent<Props> = ({ position, style }) => {
  return (
    <View style={[styles.container, style]}>
      <Text style={styles.geolocationText}>
        <Text style={styles.coordinateLabel}> Lat </Text>
        {position?.latitude}
      </Text>
      <Text style={styles.geolocationText}>
        <Text style={styles.coordinateLabel}> Lon </Text>
        {position?.longitude}
      </Text>
      <Text style={styles.geolocationText}>
        <Text style={styles.coordinateLabel}> Alt </Text>
        {position?.altitude}
      </Text>
    </View>
  )
}

export default CoordinatesView
