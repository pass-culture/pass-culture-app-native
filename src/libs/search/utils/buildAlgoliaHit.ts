import { ResultItem } from '@elastic/app-search-javascript'

import { CategoryNameEnum } from 'api/gen'
import { AlgoliaHit } from 'libs/algolia'
import { SuggestedPlace } from 'libs/place'
import { AppSearchFields, TRUE } from 'libs/search/filters/constants'

// TODO(antoinewg) We need this function temporarily but delete when we migrate completely to App Search
export const buildAlgoliaHit = (searchHit: ResultItem<AppSearchFields>): AlgoliaHit => {
  const dates = (searchHit.getRaw(AppSearchFields.dates) as string[]).map(
    (ts: string) => new Date(ts).getTime() / 1000
  )
  const prices = (searchHit.getRaw(AppSearchFields.prices) as string[]).map((p: string) => +p / 100)
  const geoloc = searchHit.getRaw(AppSearchFields.venue_position) as string
  const [lat, lng] = (geoloc || ',').split(',')

  return {
    offer: {
      category: searchHit.getRaw(AppSearchFields.category) as CategoryNameEnum,
      dates,
      description: searchHit.getRaw(AppSearchFields.description) as string,
      isDigital: +searchHit.getRaw(AppSearchFields.is_digital) === TRUE,
      isDuo: +searchHit.getRaw(AppSearchFields.is_duo) === TRUE,
      name: searchHit.getRaw(AppSearchFields.name) as string,
      prices,
      thumbUrl: searchHit.getRaw(AppSearchFields.thumb_url) as string,
    },
    _geoloc: {
      lat: isNaN(parseFloat(lat)) ? null : parseFloat(lat),
      lng: isNaN(parseFloat(lng)) ? null : parseFloat(lng),
    },
    objectID: searchHit.getRaw(AppSearchFields.id) as string,
  }
}

export const buildVenues = (searchHit: ResultItem<AppSearchFields>): SuggestedPlace => {
  const geoloc = searchHit.getRaw(AppSearchFields.venue_position) as string
  const [lat, lng] = (geoloc || ',').split(',')
  const latitude = isNaN(parseFloat(lat)) ? null : parseFloat(lat)
  const longitude = isNaN(parseFloat(lng)) ? null : parseFloat(lng)

  const venueId = searchHit.getRaw(AppSearchFields.venue_id) as string

  return {
    label: searchHit.getRaw(AppSearchFields.venue_name) as string,
    info: searchHit.getRaw(AppSearchFields.offerer_name) as string,
    geolocation: latitude && longitude ? { latitude, longitude } : null,
    venueId: isNaN(parseFloat(venueId)) ? null : parseFloat(venueId),
  }
}
