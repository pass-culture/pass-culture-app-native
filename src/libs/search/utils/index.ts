export { filterAvailableCategories } from './filterAvailableCategories'
export { getCategoriesFacetFilters } from './getCategoriesFacetFilters'
