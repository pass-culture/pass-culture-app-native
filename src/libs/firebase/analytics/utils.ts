import { NativeSyntheticEvent, NativeScrollEvent } from 'react-native'

import { SearchState } from 'features/search/types'
import { PerformSearchState } from 'libs/firebase/analytics/types'

type Props = NativeSyntheticEvent<NativeScrollEvent>['nativeEvent'] & { padding?: number }

export const isCloseToBottom = ({
  layoutMeasurement,
  contentOffset,
  contentSize,
  padding = 20,
}: Props) => layoutMeasurement.height + contentOffset.y >= contentSize.height - padding

export const isCloseToEndHorizontal = ({
  layoutMeasurement,
  contentOffset,
  contentSize,
  padding = 0,
}: Props) => layoutMeasurement.width + contentOffset.x >= contentSize.width - padding

// We don't send integers to firebase because they will be cast into int_value, float_value,
// or double_value in BigQuery depending on its value. To facilitate the work of the team,
// we just cast it to string.
export const prepareLogEventParams = (params: Record<string, unknown>) =>
  Object.keys(params).reduce((acc: Record<string, unknown>, key) => {
    acc[key] = typeof params[key] === 'number' ? (params[key] as number).toString() : params[key]
    return acc
  }, {})

const STRING_VALUE_MAX_LENGTH = 100
export const urlWithValueMaxLength = (url: string) => url.slice(0, STRING_VALUE_MAX_LENGTH)

export const buildPerformSearchState = (searchState: SearchState) => {
  const state: PerformSearchState = {
    searchLocationFilter: JSON.stringify(searchState.locationFilter),
    searchId: searchState.searchId,
  }

  if (searchState.date !== null) {
    state.searchDate = JSON.stringify(searchState.date)
  }

  if (searchState.isAutocomplete) {
    state.searchIsAutocomplete = searchState.isAutocomplete
  }

  if (searchState.maxPrice) {
    state.searchMaxPrice = searchState.maxPrice
  }

  if (searchState.minPrice) {
    state.searchMinPrice = searchState.minPrice
  }

  if (searchState.offerCategories.length > 0) {
    state.searchCategories = JSON.stringify(searchState.offerCategories)
  }

  if (searchState.offerGenreTypes && searchState.offerGenreTypes.length > 0) {
    state.searchGenreTypes = JSON.stringify(searchState.offerGenreTypes)
  }

  if (searchState.offerIsDuo) {
    state.searchOfferIsDuo = searchState.offerIsDuo
  }

  if (searchState.offerIsFree) {
    state.searchOfferIsFree = searchState.offerIsFree
  }

  if (searchState.offerNativeCategories && searchState.offerNativeCategories.length > 0) {
    state.searchNativeCategories = JSON.stringify(searchState.offerNativeCategories)
  }

  if (searchState.query !== '') {
    state.searchQuery = searchState.query
  }

  if (searchState.timeRange !== null) {
    state.searchTimeRange = JSON.stringify(searchState.timeRange)
  }

  return state
}
