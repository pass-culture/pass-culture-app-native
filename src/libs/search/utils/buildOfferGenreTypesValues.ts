import { OfferGenreType } from 'features/search/types'

export const buildOfferGenreTypesValues = ({
  movieGenres,
  musicTypes,
  showTypes,
}: {
  movieGenres: OfferGenreType[] | undefined
  musicTypes: OfferGenreType[] | undefined
  showTypes: OfferGenreType[] | undefined
}): OfferGenreType[] => {
  let finalOfferGenreType: OfferGenreType[] = []

  finalOfferGenreType = finalOfferGenreType?.concat(movieGenres ?? [])
  finalOfferGenreType = finalOfferGenreType?.concat(musicTypes ?? [])
  finalOfferGenreType = finalOfferGenreType?.concat(showTypes ?? [])

  return finalOfferGenreType
}
