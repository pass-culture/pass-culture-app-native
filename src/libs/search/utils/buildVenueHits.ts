import { ResultItem } from '@elastic/app-search-javascript'

import { VenueHit } from 'libs/search'
import { AppSearchVenuesFields } from 'libs/search/filters/constants'

export const buildVenueHits = (
  searchHit: ResultItem<AppSearchVenuesFields>
): Pick<VenueHit, 'id' | 'name' | 'venueType'> => {
  return {
    id: searchHit.getRaw(AppSearchVenuesFields.id) as string,
    name: searchHit.getRaw(AppSearchVenuesFields.name) as string,
    venueType: searchHit.getRaw(AppSearchVenuesFields.venue_type) as string,
  }
}
