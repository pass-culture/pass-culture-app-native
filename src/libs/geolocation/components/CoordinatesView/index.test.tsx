import React from 'react'

import { render } from 'tests/utils'

import CoordinatesView from './index'

describe('CoordinatesView component', () => {
  it('should render correctly', () => {
    const coordinatesView = render(
      <CoordinatesView
        position={{
          latitude: 45,
          longitude: 25,
          altitude: 200,
          accuracy: 1,
          heading: 0,
          speed: 0,
          altitudeAccuracy: 1,
        }}
      />
    )
    expect(coordinatesView).toMatchSnapshot()
  })
})
