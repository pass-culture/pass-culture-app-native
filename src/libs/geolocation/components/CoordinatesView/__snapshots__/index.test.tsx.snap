// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoordinatesView component should render correctly 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "row",
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "fontSize": 9,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0066ff",
        }
      }
    >
       Lat 
    </Text>
    45
  </Text>
  <Text
    style={
      Object {
        "fontSize": 9,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0066ff",
        }
      }
    >
       Lon 
    </Text>
    25
  </Text>
  <Text
    style={
      Object {
        "fontSize": 9,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#0066ff",
        }
      }
    >
       Alt 
    </Text>
    200
  </Text>
</View>
`;
